### PR TITLE
fix(python): fixed panic when scan .egg archive

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description
If there are no package information files in the Python `*.egg` archive, a panic error occurs: 
```
➜   go run cmd/fanal/main.go fs /home/dmitriy/work/temp/issue1930/ 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xd4e9ea]

goroutine 50 [running]:
github.com/aquasecurity/fanal/analyzer/language/python/packaging.packagingAnalyzer.Analyze({}, {0x1905120?, 0x0?}, {{0x7fff59cd1fa8, 0x21}, {0xc0008a8be2, 0x17}, {0x1d9ab48, 0xc0002c60d0}, {0x1d95b48, ...}, ...})
        /home/dmitriy/work/fanal/analyzer/language/python/packaging/packaging.go:53 +0x18a
github.com/aquasecurity/fanal/analyzer.AnalyzerGroup.AnalyzeFile.func1({0x1d96e10, 0x2b1b540}, {0x1d98188?, 0xc000010738})
        /home/dmitriy/work/fanal/analyzer/analyzer.go:253 +0x253
created by github.com/aquasecurity/fanal/analyzer.AnalyzerGroup.AnalyzeFile
        /home/dmitriy/work/fanal/analyzer/analyzer.go:248 +0x3d8
exit status 2
``` 
This is because we are trying to [Close()](https://github.com/aquasecurity/fanal/blob/3c984c398d36dcca6a78d9a0a64d7c92610dbcda/analyzer/language/python/packaging/packaging.go#L53) `nil` value.

Added error for `*.egg` archives without package information files.:
```
➜   go run cmd/fanal/main.go fs /home/dmitriy/work/temp/issue1930/
WARN: unknown OS
/home/dmitriy/work/temp/issue1930
RepoTags: []
RepoDigests: []
<nil>
via image Packages: 0
```

## Related issues
- aquasecurity/trivy/issues/1930

### CODE REVIEW

**Comment by knqyf263** [comment]
@DmitriyLewen @AndreyLevchenko any updates?

**Review by AndreyLevchenko — APPROVED** [review]

File: analyzer/language/python/packaging/packaging.go [hunk 1]
```diff
@@ -50,7 +50,16 @@ func (a packagingAnalyzer) Analyze(_ context.Context, input analyzer.AnalysisInp
...
-		defer pkginfoInZip.Close()
+		defer func() {
+			if pkginfoInZip != nil {
+				pkginfoInZip.Close()
+			}
+		}()
+
+		if pkginfoInZip == nil { // Egg archive may not contain required files, then we will get nil. Skip this archives
+			return nil, nil
+		}
```
Comment (knqyf263) [inline analyzer/language/python/packaging/packaging.go:61]: Can we return early?

```suggestion
		if pkginfoInZip == nil { // Egg archive may not contain required files, then we will get nil. Skip this archives
			return nil, nil
		}

		defer pkginfoInZip.Close()
```

Comment (DmitriyLewen) [reply analyzer/language/python/packaging/packaging.go]: Looks better, Thanks! looks better, Thanks!

Comment (knqyf263) [reply analyzer/language/python/packaging/packaging.go]: You said twice😄

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**